src/Features/CSharp/Portable/Completion/CompletionProviders/InternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,145 @@
+                if (node is AttributeArgumentSyntax)
+                {
+                    node = node.Parent;
+                    if (node is AttributeArgumentListSyntax)
+                    {
+                        return node.Parent as AttributeSyntax;
+                    }
+                }
+            }
+            return default(AttributeSyntax);
+        }
+
+        private static void AddAssemblyCompletionItems(CompletionContext context, CancellationToken cancellationToken)
+        {
+            var currentProject = context.Document.Project;
+            foreach (var p in context.Document.Project.Solution.Projects)
+            {
+                if (p == currentProject)
+                {
+                    continue;
+                }
+                var completionItem = CompletionItem.Create(displayText: p.AssemblyName,
+                    properties: ImmutableDictionary.Create<string, string>().Add(ProjectIdKey, p.Id.Id.ToString()));
+                context.AddItem(completionItem);
+            }
+        }
+
+        public override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey = default(char?), CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (item.Properties.TryGetValue(ProjectIdKey, out var projectId))
+            {
+                var assemblyName = item.DisplayText;
+                var project = document.Project.Solution.GetProject(ProjectId.CreateFromSerialized(new System.Guid(projectId)));
+                var publicKey = await GetPublicKeyOfProjectAsync(project, cancellationToken).ConfigureAwait(false);
+                if (!string.IsNullOrEmpty(publicKey))
+                {
+                    assemblyName += ", PublicKey=" + publicKey;
+                }
+                var tc = new TextChange(item.Span, assemblyName);
+                return CompletionChange.Create(tc);
+            }
+            Debug.Fail("Project can't be found by projectId.");
+            return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<string> GetPublicKeyOfProjectAsync(Project p, CancellationToken cancellationToken)
+        {
+            var compilationOptions = p.CompilationOptions;
+            if (compilationOptions == null ||
+                (string.IsNullOrEmpty(compilationOptions.CryptoKeyFile) && string.IsNullOrEmpty(compilationOptions.CryptoKeyContainer)))
+            {
+                return string.Empty;
+            }
+            var c = await p.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            if (c.Assembly?.Identity?.IsStrongName ?? false)
+            {
+                return GetPublicKeyAsHexString(c.Assembly.Identity.PublicKey);
+            }
+            return string.Empty;
+        }
+
+        private static string GetPublicKeyAsHexString(ImmutableArray<byte> publicKey)
```

Comment:
This method is a duplication of code from [AssemblyIdentity.DisplayName.cs](https://github.com/dotnet/roslyn/blob/65cc61578e9646cf76a297d8a9e0005afa57378a/src/Compilers/Core/Portable/MetadataReference/AssemblyIdentity.DisplayName.cs#L88). Maybe this can be refactored to a common extension method? For alternative implementations see also [Stackoverflow](https://stackoverflow.com/questions/311165/how-do-you-convert-byte-array-to-hexadecimal-string-and-vice-versa) or [Stackoverflow](https://stackoverflow.com/questions/623104/byte-to-hex-string/623114)